### PR TITLE
feat(ENG-10388): add lib for chromia native integration

### PR DESCRIPTION
**Pull request by thakursachin467**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,243 @@
...
+```typescript
+export type Network = {
+  networkName: string;
+  transactionLink: (arg0: string) => string;
+  walletLink: (arg0: string) => string;
+  networkUrl: string;
+};
+```
+
+- `networkName`: The name of the network (e.g., `mainnet`, `testnet`).
```
Comment (shivamangina) [inline README.md]: change it `main` and `testnet`

File: test/chromia.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
+import ChromiaLib from '../src/index';
```
Comment (shivamangina) [inline test/chromia.ts]: file name should be `chromia.test.ts`

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+/**
+ * Validate the wallet address
+ * @param address
+ * @returns
+ */
+const isValidWalletAddress = async (address: string) => {
+  // validate chromia address
+  try {
+    const hashRegex = /^[a-fA-F0-9]+$/;
```
Comment (shivamangina) [inline src/index.ts:41]: On chromia we will only be able to send the assets if the user wallet is a `Registered account` , We also need to check if the user is registered account or not and throw the error

Comment (thakursachin467) [reply src/index.ts:41]: @shivamangina We can send tokens even if an account is not registered. Users will get an option to create the account when they go to use those funds. And only the economy chain has this requirement. Not all chains inside chromia will work this way. Please find the example below where 10tCHR were transfered to an account which is not registered

image

File: test/chromia.ts (hunk 2 above)
Comment (thakursachin467) [reply test/chromia.ts]: Resolved Please check

File: README.md (hunk 1 above)
Comment (thakursachin467) [reply README.md]: Resolved Please check

**Review by shivamangina — APPROVED** [review]

**Pull request merged**